### PR TITLE
Document agent-managed-entity setting

### LINKED ISSUE #2905: Document the new agent-managed-entity configuration attribute
### Description
We will add a new configuration attribute in 6.2.0 for sensu-agent named `agent-managed-entity`, which allows an agent to manage its entity, rather than via the backend API. This is the same behavior we had in pre-6.0 agents.

See https://github.com/sensu/sensu-go/pull/4132.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
- https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/agent/
- https://docs.sensu.io/sensu-go/latest/api/entities/
- ?

### Expected behavior
When an agent is started with the flag `--agent-managed-entity` or `agent-managed-entity: true`, it becomes responsible for managing its entity configuration.

This means the entity will no longer be able to be edited via the backend API (we will return HTTP 409 Conflict in such case).

To update its configuration, an agent simply needs to be restarted, which is the same behavior we had with sensu-agent 5.x.

### Current behavior
In Sensu Go 6.x, only the initial configuration of an agent is used for its entity; once registered, this entity can only be modified via the backend API.

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds agent-managed-entity flag and description table to agent reference
- Adds information about using agent management for agent entities in the entities reference

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2905

### CODE REVIEW

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 1]
```diff
@@ -771,6 +771,7 @@ Usage:
   sensu-agent start [flags]
 
 Flags:
+      --agent-managed-entity                  manage this entity via the agent
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md:774]: I do not see this flag listed when I run `sensu-agent start --help`

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md [hunk 2]
```diff
@@ -837,6 +838,22 @@ See the [example agent configuration file][5] (also provided with Sensu packages
...
+
+| agent-managed-entity |      |
+-------------|------
+description  | Indicates whether the agent's entity is solely managed by the agent rather than the backend API.
+required     | false
+type         | Boolean
+default      | false
+environment variable | `SENSU_AGENT_MANAGED_ENTITY`
+example       | {{< code shell >}}# Command line example
+sensu-agent start --agent-managed-entity
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md:851]: I get an error when I run this command:
```
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % sensu-agent version
sensu-agent version (devel)+ce, community edition, built with go1.13.3
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % sensu-agent start --agent-managed-entity
{"component":"agent","error":"unknown flag: --agent-managed-entity","level":"fatal","msg":"error executing sensu-agent","time":"2020-12-15T16:00:40-05:00"}
hillaryfraley@Hillarys-Sensu-MacBook-Pro sensu-enterprise-go % 
```

File: content/sensu-go/6.2/observability-pipeline/observe-entities/entities.md [hunk 3]
```diff
@@ -26,23 +26,34 @@ See [the announcement on our blog][11] for more information about our usage poli
 When an agent connects to a backend, the agent entity definition is created from the information in the `agent.yml` configuration file.
 The default `agent.yml` file location [depends on your operating system][35].
 
+### Manage agent entities via the backend
+
 You can manage agent entities via the backend with [sensuctl][37], the [entities API][36], and the [web UI][33], just like any other Sensu resource.
 This means you do not need to update the `agent.yml` configuration file to add, update, or delete agent entity attributes like subscriptions and labels.
+This is the default configuration for agent entities.
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-entities/entities.md:33]: I just took a guess that this configuration is still an option, and that it is the default.

**Comment by hillaryfraley** [comment]
@palourde I could not seem to generate an agent entity with the agent-managed-entity attribute included in its definition, so I am not sure what updates I should make to the entity API doc.

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md (hunk 1 above)
Comment (palourde) [reply content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md:774]: Make sure to pull the latest changes and rebuild your agent:

```
$ sensu-agent -h
start the sensu agent

Usage:
  sensu-agent start [flags]

Flags:
      --agent-managed-entity                manage this entity via the agent
[...]
```

File: content/sensu-go/6.2/observability-pipeline/observe-entities/entities.md (hunk 3 above)
Comment (palourde) [reply content/sensu-go/6.2/observability-pipeline/observe-entities/entities.md:33]: That's right; this new behavior is opt-in, and the default is still using the backend to manage entities.

**Comment by palourde** [comment]
@hillaryfraley If you try to update an entity managed by its agent via sensuctl, you will receive the following error:

```
$ sensuctl create -f bin/gin.yml
Error: error putting resource #0 with name "gin" and namespace "default" (/api/core/v2/namespaces/default/entities/gin): entity is managed by its agent
```

And if you use curl:

```
$ curl -v -X PUT \
-H "Authorization: Key $SENSU_API_KEY" \
-H 'Content-Type: application/json' \
-d '{
  "metadata": {
    "namespace": "default",
    "name": "gin"
  },
  "entity_class": "agent"
}
' \
http://127.0.0.1:8080/api/core/v2/namespaces/default/entities/gin
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> PUT /api/core/v2/namespaces/default/entities/gin HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.64.1
> Accept: */*
> Authorization: Key d0be75ca-13eb-456a-bae2-43f147db3c8a
> Content-Type: application/json
> Content-Length: 97
>
* upload completely sent off: 97 out of 97 bytes
< HTTP/1.1 409 Conflict
< X-Content-Type-Options: nosniff
< Date: Tue, 15 Dec 2020 23:33:21 GMT
< Content-Length: 53
< Content-Type: text/plain; charset=utf-8
<
* Connection #0 to host 127.0.0.1 left intact
{"message":"entity is managed by its agent","code":3}* Closing connection 0
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md (hunk 1 above)
Comment (palourde) [reply content/sensu-go/6.2/observability-pipeline/observe-schedule/agent.md:774]: Also, it's probably only available on sensu-go at the moment, not sensu-enterprise-go yet.

**Review by palourde — APPROVED** [review]
👍

**Comment by hillaryfraley** [comment]
@palourde I made a few updates and additions based on your clarifications above. Would you mind taking another look to double-check these further changes?

**Review by palourde — APPROVED** [review]
Latest commit is perfect, thanks for adding that clarification!

**Pull request merged**